File: demos/W9D1-js-prototypes/webpack/cat.js
const Animal = require("./animal.js");

function Cat(name) {
  this.name = name;
}

function Surrogate() { }
Surrogate.prototype = Animal.prototype;
Cat.prototype = new Surrogate();
Cat.prototype.constructor = Cat;

Cat.prototype.meow = function () {
  console.log(`meow`);
};

// let pinky = new Cat(`pinky`);

module.exports = Cat;

File: lectures/w4d2-oop/demos/animals/cat.rb
require_relative 'animal'

class Cat < Animal
    def initialize(name, fur_color)
        super(name)
        @fur_color = fur_color
    end

    def introduce
        puts "#{name} meow"
    end

    # def eat(food)
    #     puts "*#{name} eats the #{food}*"
    # end

    private
    attr_reader :name
end